pkg/engine/virtualmachinescalesets.go

```diff
@@ -611,6 +611,11 @@ func CreateAgentVMSS(cs *api.ContainerService, profile *api.AgentPoolProfile) Vi
 			CustomData: to.StringPtr(customDataStr),
 		}
 		vmssVMProfile.OsProfile = &windowsOsProfile
+
+		if cs.Properties.WindowsProfile.GetEnableAHUB() {
```

Comment:
may WindowsProfile be nil here?

Reply:
No. Since above code has checked the agent pool is Windows.